src/components/MAPI/utils.ts

```diff
@@ -405,6 +406,71 @@ export function getProviderNodePoolLocation(
   }
 }
 
+// TODO(axbarsan): Get this info from the environment, rather than the info response.
+export function getSupportedAvailabilityZones(
+  state: IState
+): {
+  minCount: number;
+  maxCount: number;
+  defaultCount: number;
+  all: string[];
+} {
+  const zonesStats = state.main.info.general.availability_zones;
+
+  return {
+    minCount: 1,
+    maxCount: zonesStats?.max ?? 1,
+    defaultCount: zonesStats?.default ?? 1,
+    all: zonesStats?.zones ?? [],
+  };
+}
+
+/**
+ * Determine a random list list of availability zones,
+ * based on a given collection of available zones and
+ * a required number of elements.
+ * @param count - The maximum number of zones.
+ * @param fromList - The available zones.
+ * @param existing - Some zones that must be included in the output.
+ */
+export function determineRandomAZs(
```

Comment:
This function generates a random list of AZs, taking into account the desired number of zones, and the list of supported zones.

If the AZs of the control plane nodes are supplied, then the list is not that random, as those zones would be picked first.

For example, if the cluster has 1 control plane node in the `2` AZ, and we want to have a node pool with 3 AZs, and all the supported zones are `['1', '2', '3', '4']`, then the result would probably be `['1', '2', '3']`, `['2', '3', '4']` or other combination of the rest.

If you want to have a more detailed explanation of the behaviour, please check the tests for this function.